clients/scheduler/src/components/time-slot-select/index.tsx

```diff
@@ -122,10 +123,13 @@ export const TimeSlotSelect = () => {
           ...rest,
           provider: findProvider(providerId, providers),
         })),
-        isFirstDateViewed:
-          previousInitializedValue === false && initialized === true,
+        isFirstDateViewed: datesViewed === 0,
+        datesViewed,
```

Comment:
Do we need to pass in datesViewed into this callback? If so, do we need `isFirstDateViewed`? We could just look for `datesViewed === 0` in the scheduler and then we wouldn't need `isFirstDateViewed`

Reply:
I just added datesViewed so we can use it if we want, i agree it makes `isFirstDateViewed` less relevant, but seemed like in the context of a bug fix i shouldn't go add new functionality to parse / make sense of this in jade.

Reply:
gotcha. I was thinking in terms of expected type schema of the object arg on the scheduler embed end. I know we have a million TS issues there, but I'd like to not add to them if we can. If we add `datesViewed` here, would we need to explicitly add it to the expected callback arg type for the Canvas callback? I'm guessing it's called with type `any`, in which case this is fine. Is that the case?